async_tcp_server.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef ASYNC_TCP_SERVER
+#define ASYNC_TCP_SERVER
+
+#include "socket_wrapper.hpp"
+#include <vector>
+
+using Socket_vector = std::vector<Socket_sptr>;
+
+class AsyncTCPServer
+{
+    fd_set                          m_fdsetRead;
+
+    Socket_vector                   m_aSockets;
+
+    public:
+
+                                    AsyncTCPServer();
+                                    AsyncTCPServer(Socket_sptr masterSocket, fd_set fdsetRead);
+    virtual                         ~AsyncTCPServer();
```

Comment:
Why the distructor is virtual here? It's not too bad, but create little overhead because of vtable and vpointers.